### DISCUSSION TITLE
Any way to force particular users to run firejail without options or profile overrides?

### DISCUSSION THREAD

**Discussion by jonleivent** [netblue30/firejail#5839] [Q&A]
I am considering writing my own suid exe that calls firejail without options, passing just the app name and its options, thus forcing the user to use only an existing profile.  But I would also like the ability to disallow the user's own profile overrides in ~/.config/firejail as well.  Is there any way to accomplish this, other than patching the firejail sources?

**Comment by rusty-snake** [comment]
You can configure firejail with`--enable-only-syscfg-profiles` to only load profiles from the system path.

If you then disallow calling firejail directly, there should be no way to change the profile I guess.

---

What's your treat model? Why do you want this?

Remember that users can still call programs by path and completely avoid firejail.

**Reply by jonleivent** [reply]
> What's your treat model? Why do you want this?
> Remember that users can still call programs by path and completely avoid firejail.

I'm using AppArmor to prevent this particular user from avoiding firejail for certain apps, or bypassing system profiles for those apps.  The point is to create a very secure desktop for this particular user that can only run a small curated set of apps and that can't be hijacked to run anything else either accidentally or intentionally.  The entire desktop runs with AppArmor profiles to force this.  I could use AppArmor to block reads and writes of `~/.config/firejail/*` but don't know if that will cause problems with firejail, as that would require starting firejail with an attached AppArmor profile.  Or I could modify firejail's own AppArmor profile, but I don't know when it reads from `~/.config/firejail` vs. changes into that profile.  Or I could use AppArmor to force the entire desktop to not have write access to `~/.config/firejail` so that nothing can be put there.  I'm just starting to explore all of these options.

Will `--enable-only-syscfg-profiles` also avoid loading include files from `~/.config/firejail`?  Can I still force an absolute pathname with the `--profile=...` option?  Also, I need a way other users can run firejail without this restriction, so that means having multiple firejails at different paths or with different names.
